packages/stores/src/account/__tests_e2e__/positions.spec.ts

```diff
@@ -0,0 +1,171 @@
+        1,
+        0.001, // must have spread factor to generate fees
+        undefined,
+        (tx) => {
+          if (tx.code) reject(tx.log);
+          else resolve(tx);
+        }
+      )
+    );
+
+    queryPool = await getLatestQueryPool(chainId, queriesStore);
+  });
+
+  it("should be able to be created", async () => {
+    await expect(createFullRangePosition(queryPool!.id)).resolves.toBeDefined();
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(1);
+    await expect(createFullRangePosition(queryPool!.id)).resolves.toBeDefined();
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(2);
+  });
+
+  it("can have liquidity be added to it", async () => {
+    const account = accountStore.getAccount(chainId);
+
+    // create 2 positions, since there need to be at least 2 to be able to add liquidity
+    await createFullRangePosition(queryPool!.id);
+    await createFullRangePosition(queryPool!.id);
+    const userPositionIds = await getUserPositionsIdsForPool(queryPool!.id);
+    const lastPositionId = userPositionIds[userPositionIds.length - 1];
+
+    // add liquidity to position
+    const specifiedAmount0 = "1000";
+    const calculatedAmount1 = "1000";
+
+    await expect(
+      new Promise((resolve, reject) =>
+        account.osmosis
+          .sendAddToConcentratedLiquidityPositionMsg(
+            lastPositionId,
+            specifiedAmount0,
+            calculatedAmount1,
+            undefined,
+            undefined,
+            (tx) => {
+              if (tx.code) reject(tx.log);
+              else resolve(tx);
+            }
+          )
+          .catch(reject)
+      )
+    ).resolves.toBeDefined();
+
+    // old position is replaced
+    await expect(
+      getUserPositionsIdsForPool(queryPool!.id)
+    ).resolves.toHaveLength(2);
+  });
+
+  it("rejects adding liquidity to position if last position in pool", async () => {
+    const account = accountStore.getAccount(chainId);
+
+    // create initial position
+    await createFullRangePosition(queryPool!.id);
+    const userPositionIds = await getUserPositionsIdsForPool(queryPool!.id);
+    const lastPositionId = userPositionIds[userPositionIds.length - 1];
+
+    // add liquidity to position
+    const specifiedAmount0 = "1000";
+    const calculatedAmount1 = "1000";
+
+    await expect(
+      new Promise((resolve, reject) =>
+        account.osmosis
+          .sendAddToConcentratedLiquidityPositionMsg(
+            lastPositionId,
+            specifiedAmount0,
+            calculatedAmount1,
+            undefined,
+            undefined,
+            (tx) => {
+              if (tx.code) reject(tx.log);
+              else resolve(tx);
+            }
+          )
+          .catch(reject)
+      )
+    ).rejects.toBeDefined();
+  });
+
+  /** Leave `poolId` undefined to get all position IDs. */
+  async function getUserPositionsIdsForPool(poolId?: string) {
```

Comment:
A general thought on this test file. The poolId is optional. Is it possible that the pool passed in here or the below function won't exist? Should we write a test for that case?

Reply:
That's moreso a utility for letting a test either get all user positions (undefined) or all user positions filtered for a pool (poolId="2"). I can remove optionality to make things less variable in this test env?

Reply:
Yea I think that makes sense 👍